engine/src/flutter/lib/web_ui/lib/src/engine/platform_dispatcher.dart

```diff
@@ -1013,6 +1022,117 @@ class EnginePlatformDispatcher extends ui.PlatformDispatcher {
+    )?.toDouble();
+    final double? letterSpacing = parseStyleProperty(
+      _typographyMeasurementElement!,
+      'letter-spacing',
+    )?.toDouble();
+    // There is no direct CSS property for paragraph spacing,
+    // so on the web this feature is usually implemented
+    // by extension authors by leveraging `margin-bottom` on
+    // the `p` element.
+    final double? paragraphSpacing = parseStyleProperty(
+      _typographyMeasurementElement!,
+      'margin-bottom',
+    )?.toDouble();
+    if (lineHeight == null &&
+        wordSpacing == null &&
+        letterSpacing == null &&
+        paragraphSpacing == null) {
+      return null;
+    }
+    return ui.TypographySettings(
+      lineHeight: lineHeight,
+      letterSpacing: letterSpacing,
+      wordSpacing: wordSpacing,
+      paragraphSpacing: paragraphSpacing,
+    );
+  }
+
+  /// Set the callback function for updating [typographySettings] based on
+  /// the sizing changes of an off-screen element with text.
+  void _addTypographySettingsObserver() {
+    _typographyMeasurementElement = createDomHTMLParagraphElement();
+    _typographyMeasurementElement!.text = 'flutter typography measurement';
+    // The element should be hidden from screen readers.
+    _typographyMeasurementElement!.setAttribute('aria-hidden', 'true');
+    final DomCSSStyleDeclaration style = _typographyMeasurementElement!.style;
+    style.position = 'absolute';
+    // The element should be off-screen and not visible.
+    style.top = '-9999px';
+    style.left = '-9999px';
+    style.visibility = 'hidden';
+    // The element should be sensitive to letter-spacing, word-spacing,
+    // and line-height changes.
+    style.width = 'auto';
+    style.height = 'auto';
+    style.whiteSpace = 'nowrap';
+    // Set text spacing properties defaults.
+    const double spacingDefault = 100.0;
+    style.lineHeight = '${spacingDefault}px';
+    style.letterSpacing = '${spacingDefault}px';
+    style.wordSpacing = '${spacingDefault}px';
+    style.margin = '0px 0px ${spacingDefault}px 0px';
+    domDocument.body!.append(_typographyMeasurementElement!);
+
+    _typographySettingsObserver = createDomResizeObserver((
+      List<DomResizeObserverEntry> entries,
+      DomResizeObserver observer,
+    ) {
+      final ui.TypographySettings? computedTypographySettings = _computeTypographySettings();
+      if (computedTypographySettings != null &&
+          computedTypographySettings.lineHeight == spacingDefault &&
+          computedTypographySettings.wordSpacing == spacingDefault &&
+          computedTypographySettings.letterSpacing == spacingDefault &&
+          computedTypographySettings.paragraphSpacing == spacingDefault) {
+        // Ignore initial resize event if computed values match default ones.
+        if (typographySettings == null) {
+          return;
+        }
+        // Disable text spacing properties.
+        _updateTypographySettings(const ui.TypographySettings());
```

Comment:
I'm not a fan of setting this to an empty `TypographySettings` to disable this but I don't see another way since `configuration.copyWith(typographySettings:null)` won't work.